media-api/app/controllers/SearchParams.scala

```diff
@@ -0,0 +1,161 @@
+package controllers
+
+import com.gu.mediaservice.lib.auth.{Authentication, Tier}
+import com.gu.mediaservice.lib.formatting.{parseDateFromQuery, printDateTime}
+import lib.querysyntax.{Condition, Parser}
+import org.joda.time.DateTime
+import scalaz.syntax.applicative._
+import scalaz.syntax.std.list._
+import scalaz.syntax.validation._
+import scalaz.{Validation, ValidationNel}
+
+import scala.util.Try
+
+case class SearchParams(
+  query: Option[String],
+  structuredQuery: List[Condition],
+  ids: Option[List[String]],
+  offset: Int,
+  length: Int,
+  orderBy: Option[String],
+  since: Option[DateTime],
+  until: Option[DateTime],
+  modifiedSince: Option[DateTime],
+  modifiedUntil: Option[DateTime],
+  takenSince: Option[DateTime],
+  takenUntil: Option[DateTime],
+  archived: Option[Boolean],
+  hasExports: Option[Boolean],
+  hasIdentifier: Option[String],
+  missingIdentifier: Option[String],
+  valid: Option[Boolean],
+  free: Option[Boolean],
+  payType: Option[PayType.Value],
+  hasRightsCategory: Option[Boolean],
+  uploadedBy: Option[String],
+  labels: List[String],
+  hasMetadata: List[String],
+  persisted: Option[Boolean],
+  usageStatus: List[String],
+  usagePlatform: List[String],
+  tier: Tier
+)
+
+case class InvalidUriParams(message: String) extends Throwable
+object InvalidUriParams {
+  val errorKey = "invalid-uri-parameters"
+}
+
+object PayType extends Enumeration {
+  type PayType = Value
+  val Free = Value("free")
+  val MaybeFree = Value("maybe-free")
+  val All = Value("all")
+  val Pay = Value("pay")
+
+  def create(s: String) = s match {
+    case "free" => Some(Free)
```

Comment:
Is it worth pulling these Strings out into private object members given we're relying on them being the same?

Reply:
I'm going to leave this file as it is because it's not directly related to this work, I've just moved the SearchParam classes out of the controller file.